components/object_store/src/obkv/mod.rs

```diff
@@ -0,0 +1,940 @@
+                source,
+            })?;
+
+        Ok(ObjectMeta {
+            location: (*location).clone(),
+            last_modified,
+            size: meta.size,
+        })
+    }
+
+    /// Delete the object at the specified location.
+    async fn delete(&self, location: &Path) -> Result<()> {
+        let instant = Instant::now();
+
+        // TODO: maybe coerruption here, should not delete data when data is reading.
+        let table_name = self.pick_shard_table(location);
+        let meta =
+            self.read_meta(location)
+                .await
+                .box_err()
+                .map_err(|source| StoreError::NotFound {
+                    path: location.to_string(),
+                    source,
+                })?;
+
+        // delete every part of data
+        for part in &meta.parts {
+            let key = part.as_bytes();
+            self.client
+                .delete(table_name, key)
+                .map_err(|source| StoreError::Generic {
+                    store: OBKV,
+                    source: Box::new(source),
+                })?;
+        }
+        // delete meta info
+        self.meta_manager
+            .delete(meta, location)
+            .await
+            .map_err(|source| StoreError::Generic {
+                store: OBKV,
+                source: Box::new(source),
+            })?;
+
+        debug!(
+            "ObkvObjectStore delete operation, location:{location}, table:{table_name}, cost:{}",
+            instant.elapsed().as_millis()
+        );
+
+        Ok(())
+    }
+
+    /// List all the objects with the given prefix.
+    /// Prefixes are evaluated on a path segment basis, i.e. `foo/bar/` is a
+    /// prefix of `foo/bar/x` but not of `foo/bar_baz/x`.
+    /// TODO: Currently this method may return lots of object meta, we should
+    /// limit the count of return ojects in future. Maybe a better
+    /// implementation is to fetch and send the list results in a stream way.
+    async fn list(&self, prefix: Option<&Path>) -> Result<BoxStream<'_, Result<ObjectMeta>>> {
+        let instant = Instant::now();
+
+        let path = Self::normalize_path(prefix);
+        let raw_metas =
+            self.meta_manager
+                .list(&path)
+                .await
+                .map_err(|source| StoreError::Generic {
+                    store: OBKV,
+                    source: Box::new(source),
+                })?;
+
+        let mut meta_list = Vec::new();
+        for meta in raw_metas {
+            meta_list.push(Ok(ObjectMeta {
+                location: Path::from(meta.location),
+                last_modified: Utc.timestamp_millis_opt(meta.last_modified).unwrap(),
+                size: meta.size,
+            }));
+        }
+
+        let iter = futures::stream::iter(meta_list.into_iter());
+        debug!(
+            "ObkvObjectStore list operation, prefix:{path}, cost:{}",
+            instant.elapsed().as_millis()
+        );
```

Comment:
It will print well by `Duration`'s debug output:
```suggestion
        debug!(
            "ObkvObjectStore list operation, prefix:{path}, cost:{:?}",
            instant.elapsed()
        );
```